app/models/lesson_plan.rb

```diff
@@ -1,14 +1,60 @@
 require_dependency "duration"
 
 class LessonPlan < ApplicationRecord
-  has_many :lesson_plan_lessons
+  has_many :lesson_plan_lessons,
+           after_remove: ->(plan, _){ plan.update_column(:pdf_file_updated_at, nil) }
+
   has_many :lessons, through: :lesson_plan_lessons
 
-  accepts_nested_attributes_for :lesson_plan_lessons, allow_destroy: true
+  accepts_nested_attributes_for :lesson_plan_lessons,
+                                allow_destroy: true
 
   validates_uniqueness_of :key
   before_validation :set_key, on: :create
 
+  mount_uploader :pdf_file, GenericUploader
+
+  def pdf
+    unless pdf_file_updated_at.try(:>=, lessons.pluck(:updated_at).max)
+      self.recreate_pdf_file
+    end
+
+    pdf_file
+  end
+
+  def recreate_pdf_file
+    controller = LessonPlansController.new
```

Comment:
I'm a little surprised to see this kind of logic in a view.  It seems like you have to jump through some hoops, having the controller call this method which then calls the controller.  Would it make sense to create a little lib class for this?

Reply:
Yes that would make sense. I created a new class in lib/pdf_template.rb to wrap this gluing together of WickedPdf/ActionController rendering. I was doing the same procedure in app/jobs/update_lesson_pdf.rb too so this ended up making the app DRYer as well. Thanks!